### DISCUSSION TITLE
WiFi Issue

### DISCUSSION THREAD

**Discussion by G0MJW** [rern/rAudio#219] [Bug Reports]
When it reboots it does not connect to the WiFi network but instead It sets up an access point. I think there is a problem with timeouts or something. This is not what I want it to do and it isn't set up to do that. It's a real pain messing around to connect up a wired network to fix it. You may recall similar problems with the previous version e6 I reported over Christmas. That was on a PI Zero Wifi which made it rather difficult to recover. Connecting to the access point each time isn't a viable solution, especially from a machine without WiFi.

**Comment by rern** [comment]
Try:
```sh
wget -O /srv/http/bash/startup.sh https://github.com/rern/rAudio-1/raw/UPDATE/srv/http/bash/startup.sh
```
Then reboot.

**Reply by G0MJW** [reply]
Thanks - that worked - but why?

**Reply by AndyFx12** [reply]
With this one-liner you have installed a new startup-script from update-branch.

Update seems to solve your problem.

regards, Andy

**Comment by mrManniManfred** [comment]
Hi,
I am using the latest updates but still I have problems with rAudio losing my wifi settings.
Sometimes only the Access point is active and no futrther wifi listed. 
After a reboot my actual wifi is listed sometimes below the access point but inactive. After activating it the Access point is removed but only as long as I reboot. 
Then the cycle continues...

Also posted my problem description here: https://www.runeaudio.com/forum/post31138.html#p31138

Regards

**Reply by rern** [reply]
It might be your Wi-Fi needs more time to connect.

Try:
`sed -i 's/1..10/1..30/' /srv/http/bash/startup.sh`
(check connection every 1 second for 30 seconds)

Then reboot.

**Reply by mrManniManfred** [reply]
Tried this config in multiple reboots.
There are actually three constellations on different reboots:

1. My wifi is connected.
2. Only Access Point is connected, my wifi is listed below but disconnected.
3. Only Access Point is connected my wifi is not listed below (Most of the times). When clicking on the magnifiying glass my wifi is listed with a green dot (meaning it is connected but it is not actually. disconnecting and reconnecting establishes the connection).

**Reply by rern** [reply]
Get Wi-Fi signal strength:
Settings > Networks > Wi-Fi - search icon - dBm value

Anything less than -66 dBm should be considered as not good enough.

The revised `startup.sh` script now checks for a successful connection every 1 second for 30 seconds before it gives up. I think this together with your DAC issue are likely not a software problem.

**Reply by mrManniManfred** [reply]
-50 dBm.

But the software behaviour seems strange as the connection seems to be established but is not really (point 3 above).

**Reply by rern** [reply]
Try today's update again. Now it forces to reconnect if the 1st attempt failed.

**Reply by mrManniManfred** [reply]
Same behavior WiFi-wise. The Hifi-Berry-DAC though seems to get recognized less often.
What is strange is, that when clicking on my network below the connected access point the connection can be established every time. Would it be possible to shift wifi-connection further to the end of the boot process maybe this might help?

**Reply by rern** [reply]
Have you tried [the latest image file](https://github.com/rern/rAudio-1/releases/tag/rAudio_1-20210227)?

**Reply by mrManniManfred** [reply]
Hi, yes I tried the latest image file.

Another funny behavior is, that only the Access Point is shown as connected sometimes and when I disconnect it (via the menu) then my network shows below and I can connect to it...

Maybe the cursor should be activated by default when the raspberry is connected via HDMI because its easier to navigate a fresh image when wifi doesn't work.

**Reply by mrManniManfred** [reply]
With 202103121 WiFi seems work so far (6 boots in a row). Thx!
Boot takes about 50 seconds.
If the HiFiBerry DAC2 HD would be recognized as MPD every boot all my problems would be fixed :).

**Comment by G0MJW** [comment]
There is a need to allow for some quite significant delays in WiFi connectivity. Not everyone has a fast network. 

 

Mike

**Comment by G0MJW** [comment]
I just tried an update and again the WiFi is not working. Same problems as before. The fix didn't appear to work. With the Zero not having an Ethernet port, and only having one USB port, which is normally to be used for the library, using ethernet is not a simple matter. Is this likely to be fixed or do I need to give up on the Zero?

Incidentally, on a PI3 it works OK. But that caused a problem when I enabled access point mode. I have several WiFi extenders and the Pi boots faster than my router. After a power cut the entire network was attempting to get its internet connection through rune audio 192.168.5.1 and of course that did not work. It took me a while to understand why all my wired connections were down - yes wired - because the network switches all looked here as well. This may be a fault of Netgear but it is not desirable, so basically, don't enable access point mode.

**Reply by rern** [reply]
Image files just have been updated.
Try a [new one](https://github.com/rern/rAudio-1#image-files) for your Zero.

**Reply by G0MJW** [reply]
I updated it this morning. Was that since then? I have practically given up on the Zero thinking it's perhaps too limited and ordered a second hand PI3 B+
